File: src/types/habit.ts
export type TimeOfDay = 'early_morning' | 'morning' | 'afternoon' | 'evening' | 'custom' | 'anytime';
export type FrequencyType = 'daily' | 'weekly' | 'custom';
export type HabitType = 'boolean' | 'measurable';

export interface Habit {
    id: string;
    user_id: string;
    name: string;
    description?: string;
    icon: string;
    color: string;
    habit_type: HabitType;
    unit?: string; // e.g., "glasses", "minutes", "pages"
    time_of_day: TimeOfDay;
    frequency_type: FrequencyType;
    frequency_days: number[]; // 0-6
    frequency_target?: number;
    reminder_time?: string;
    is_archived: boolean;
    position: number;
    created_at: string;
}

export interface Completion {
    id: string;
    habit_id: string;
    user_id: string;
    completed_at: string; // YYYY-MM-DD
    value?: number; // For measurable habits (e.g., 3 glasses of water)
    notes?: string;
    created_at: string;
}

// Re-export StreakData for convenience
export type { StreakData } from '../lib/streaks';

File: habitify_widget/lib/models/habit.dart
/// Habit model matching web app schema
class Habit {
  final String id;
  final String name;
  final String icon;
  final String color;
  final String timeOfDay;
  final List<int> frequencyDays;
  final int? frequencyTarget;

  Habit({
    required this.id,
    required this.name,
    required this.icon,
    required this.color,
    required this.timeOfDay,
    required this.frequencyDays,
    this.frequencyTarget,
  });

  factory Habit.fromJson(Map<String, dynamic> json) {
    return Habit(
      id: json['id'] as String,
      name: json['name'] as String,
      icon: json['icon'] as String? ?? '📝',
      color: json['color'] as String? ?? '#6366f1',
      timeOfDay: json['time_of_day'] as String? ?? 'anytime',
      frequencyDays: List<int>.from(
        (json['frequency_days'] as List?)?.map((d) => int.parse(d.toString())) ?? [],
      ),
      frequencyTarget: json['frequency_target'] as int?,
    );
  }
}
